routers/routes/routes.go

```diff
@@ -495,50 +495,26 @@ func RegisterRoutes(m *macaron.Macaron) {
 				return
 			}
 
-			if attach.IssueID != 0 {
-				iss, err := models.GetIssueByID(attach.IssueID)
-				if err != nil {
-					ctx.ServerError("GetAttachmentByUUID.GetIssueByID", err)
-					return
-				}
-				repo, err := models.GetRepositoryByID(iss.RepoID)
-				if err != nil {
-					ctx.ServerError("GetAttachmentByUUID.GetRepositoryByID", err)
-					return
-				}
-				if repo.IsPrivate {
-					if !ctx.IsSigned {
-						ctx.Error(http.StatusNotFound)
-						return
-					}
-					if !repo.CheckUnitUser(ctx.User.ID, ctx.User.IsAdmin, models.UnitTypeIssues) {
-						ctx.Error(http.StatusForbidden)
-						return
-					}
-				}
-			} else if attach.ReleaseID != 0 {
-				rel, err := models.GetReleaseByID(attach.ReleaseID)
-				if err != nil {
-					ctx.ServerError("GetAttachmentByUUID.GetReleaseByID", err)
+			repository, unitType, err := attach.LinkedRepository()
+			if err != nil {
+				ctx.ServerError("LinkedRepository", err)
+				return
+			}
+
+			if repository == nil {
+				ctx.Error(http.StatusNotFound)
```

Comment:
When you are creating an issue, you uploaded the attachment but hasn't saved the issue?

Reply:
Yes that the case to catch.

Reply:
This should also be the case if you upload a file and remove it from the dropzone.

Reply:
My proposed changes should fix this